lib/feature/rockets/presentation/widget/rocket_counter_item_widget.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../../../../core/constants/colors.dart';
+import '../../logic/rocket_cubit.dart';
+import '../../logic/rocket_state.dart';
+
+class RocketCounterItemWidget extends StatelessWidget {
+  final int itemsLength;
+  const RocketCounterItemWidget(this.itemsLength, {super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    ThemeData theme = Theme.of(context);
+    return BlocBuilder<RocketCubit, RocketState>(
+      buildWhen: (previous, current) {
+        return current.maybeWhen<bool>(
+          changeRocketImage: (value) => true,
+          orElse: () => false,
+        );
+      },
+      builder: (context, state) {
+        RocketCubit cubit = RocketCubit.get(context);
```

Comment:
how getting the cubit instance here inside the builder ?

Reply:
I couldn't understand a question
I am fetching (cubit instance) through

```dart
factory RocketCubit.get(context) => BlocProvider.of<RocketCubit>(context);
```

I can access (pageImageController) and (courantImageIndex) from the cubit

Through (pageImageController) I navigate to a specific page
When the user clicks on a circle, the user is taken to the designated page

Through (courantImageIndex), I distinguish the selected page with a different design

Should I get her from a different location?


<img src="https://github.com/colonal/space_x/assets/64502335/f44ef29c-f69a-4968-a525-892eb600d76e" height="300em">